program_v2/programs/sol-did/src/instructions/update.rs

```diff
@@ -0,0 +1,48 @@
+use crate::state::{DidAccount, Secp256k1RawSignature, UpdateStruct};
+use anchor_lang::prelude::*;
+
+pub fn update(
+    ctx: Context<Update>,
+    mut update_information: UpdateStruct,
+    eth_signature: Option<Secp256k1RawSignature>,
+) -> Result<()> {
+    let data = &mut ctx.accounts.did_data;
+    data.services = update_information.services;
+    data.native_controllers = update_information.native_controllers;
+    data.other_controllers = update_information.other_controllers;
+    data.initial_verification_method.flags = 0;
+    let default_alas = &data.initial_verification_method.alias;
+    let new_methods = &mut update_information.verification_methods;
+
+    for i in new_methods.clone() {
+        if i.alias == *default_alas {
+            new_methods.retain(|x| x.alias == *default_alas);
```

Comment:
For and retain are both O(n). Can you implement this by only running through the list once.